corporate/lib/stripe.py

```diff
@@ -1291,6 +1294,14 @@ def configure_fixed_price_plan(self, fixed_price: int) -> str:
             current_plan.save(update_fields=["status", "next_invoice_date"])
             return f"Fixed price {required_plan_tier_name} plan scheduled to start on {current_plan.end_date.date()}."
 
+        if sent_invoice_id is not None:
+            fixed_price_plan_params["sent_invoice_id"] = sent_invoice_id.strip()
+            Invoice.objects.create(
+                customer=customer,
+                stripe_invoice_id=sent_invoice_id.strip(),
+                status=Invoice.SENT,
+            )
```

Comment:
It could be good to, before creating the invoice object, query the Stripe API to ask whether it can find the invoice ID, just an end-to-end check against typos before storing it in the database.

Reply:
![image](https://github.com/zulip/zulip/assets/56781761/37056e93-1911-4acf-bd8a-f8f6412c4661)

Explicitly added this 'status=open' check, because I expect there's a good chance of handling multiple support request with a lot of tabs open, resulting in copy-pasting of paid/draft invoice ID.
![image](https://github.com/zulip/zulip/assets/56781761/7878ba51-6d25-4dd0-9785-6a570957c244)

Reply:
I think removing the "Request req_..." is not of much importance as this is /support page. The error is returned by stripe, which would vary depending on error. I can improve this error message presentation, if required.

Reply:
Yeah that sounds right to me.